frog/imports/client/App/TopBar.js

```diff
@@ -112,36 +111,47 @@ class LogoutMenu extends React.Component<*, *> {
 
 class TopBarController extends React.Component<
   { classes: any, history: any, location: any },
-  {}
+  { location: string }
 > {
   routes = [
-    { name: 'Graph Editor', to: '/teacher/graph' },
-    { name: 'Sessions', to: '/t' },
+    { name: 'Graph Editor', to: ['/teacher/graph'] },
+    { name: 'Sessions', to: ['/teacher/orchestration', '/t'] },
     {
       name: 'Activity Preview',
-      to: '/teacher/preview'
+      to: ['/teacher/preview']
     }
   ];
 
-  value = '/teacher/preview';
-
   constructor(props) {
     super(props);
-    const found = this.routes.filter(
-      route => props.location.pathname.indexOf(route.to) !== -1
-    )[0];
-
-    if (found !== undefined) {
-      this.value = found.to;
-    }
+    const found = this.matchLocationPrefix();
+    this.state = {
+      location: found?.name || 'Graph Editor'
```

Comment:
Nit: move to a constant to make sure this behaviour is clear

```suggestion
      location: found?.name || DEFAULT_ROUTE
```

Reply:
Can we keep this for later when we consolidate all routing variables to constants?